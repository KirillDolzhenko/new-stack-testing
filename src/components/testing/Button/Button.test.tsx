import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Button tsx", () => {
  test("Element mounted", () => {
    render(<Button />);
    const btn = screen.getByText(/button/i);
    const btnAlso = screen.getByRole("button");
    const input = screen.getByPlaceholderText(/print text/i);

    expect(btn).toBeInTheDocument();
    expect(btnAlso).toBeInTheDocument();
    expect(input).toBeInTheDocument();

    expect(btn).toMatchSnapshot();

    // screen.debug();
  });
});
