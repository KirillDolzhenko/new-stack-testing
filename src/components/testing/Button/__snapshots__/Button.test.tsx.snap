// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tsx Element mounted 1`] = `
<button>
  Button
</button>
`;
